rpc/src/services/protocol/proto_001/helpers.rs

```diff
@@ -68,168 +58,103 @@ impl RightsConstants {
     pub(crate) fn parse_rights_constants(
         context_proto_param: &ContextProtocolParam,
     ) -> Result<Self, anyhow::Error> {
-        let dynamic =
-            tezos_messages::protocol::proto_001::constants::ParametricConstants::from_bytes(
-                &context_proto_param.constants_data,
-            )?;
-        // in proto 001, the constants are hard coded but a few exceptions modifiable in the context
-        let dynamic_all = tezos_messages::protocol::proto_001::constants::ParametricConstants::create_with_default_and_merge(dynamic);
-        let fixed = tezos_messages::protocol::proto_001::constants::FIXED;
-
-        // TODO: fix unwraps()
-        Ok(RightsConstants::new(
-            dynamic_all.blocks_per_cycle().unwrap(),
-            dynamic_all.preserved_cycles().unwrap(),
-            fixed.nonce_length(),
-            dynamic_all.time_between_blocks().as_ref().unwrap().clone(),
-            dynamic_all.blocks_per_roll_snapshot().unwrap(),
-            dynamic_all.endorsers_per_block().unwrap(),
-        ))
-    }
-}
-
-/// Data from context DB for completing baking and endorsing rights
-#[derive(Debug, Clone, Getters)]
-pub struct RightsContextData {
-    /// Random seed for Tezos PRNG
-    #[get = "pub(crate)"]
-    random_seed: Vec<u8>,
-
-    /// Number of last roll so Tezos PRNG will not overflow
-    #[get = "pub(crate)"]
-    last_roll: i32,
-
-    /// List of rolls mapped to rollers contract id
-    #[get = "pub(crate)"]
-    rolls: HashMap<i32, String>,
-}
-
-impl RightsContextData {
-    /// Simple constructor to create RightsContextData
-    pub fn new(random_seed: Vec<u8>, last_roll: i32, rolls: HashMap<i32, String>) -> Self {
-        Self {
-            random_seed,
-            last_roll,
-            rolls,
+        if let Some(constatns_deserialized) =
+            serde_json::from_str::<serde_json::Value>(&context_proto_param.constants_data)?
+                .as_object()
+        {
+            let blocks_per_cycle = get_constant_i64(constatns_deserialized, "blocks_per_cycle")?;
```

Comment:
@adonagy my idea was that for these JSONs there would just be a struct defined that would use a `#[derive(Deserialize)]` annotation to handle the parsing. Is there any reason it needs to be done manually?

Reply:
That's a really good point. We should definitely use defined structs.